### ISSUE TITLE
Hook secrets from environ variable.

### ISSUE THREAD

**Issue by jamielennox** [adnanh/webhook#160]
So I'm evaluating webhook for our purposes. 

Ideally the way I would like to do this is in a kubernetes pod: 
  - webhook, obviously 
  - nginx + letsencrypt for incoming webhooks and for displaying results
  - git-sync to fetch the hooks and scripts

Most of this looks easily doable, including the hooks reloading is great. The problem i see is that this means that i'm checking my github (or whatever) secrets into git. What I'd like is that the hooks file can specify to pull that secret from an environ variable (or file as kube can make this work, but environ seems more practical).

Ideally this wouldn't be passed through to the script itself, but consider that an extra. 

Thanks.

**Comment by adnanh** [comment]
In the described case, you should evaluate the secret in the script webhook will invoke and use the webhook to simply pass the supplied parameter via environment variable or the command argument. The actual secret should also be stored in the environment variable so that you don't have to check in the secret to the git.

**Comment by jamielennox** [comment]
I see that I can do that but by evaulating the secret in the script i miss out on being able to use the existing filters such as payload-hash-sha1 as described: https://github.com/adnanh/webhook/wiki/Hook-Examples#incoming-github-webhook or any other form of authenticated webhook.

My reasoning to use webhook rather than simply execute shell scripts out of apache or something is because it can generically handle this sort of validation for me.

**Comment by moorereason** [comment]
@adnanh,

What if the hooks definition file could be evaluated as a [Go template](https://golang.org/pkg/text/template/)?

```
webhook -hooks hooks.json -template
```

Then create a simple `getenv` function for the template FuncMap that uses `os.Getenv`.

Sample template usage:
```handlebars
"match":
{
  "type": "payload-hash-sha1",
  "secret": "{{ getenv "GITHUB_SECRET" }}",
  "parameter":
  {
    "source": "header",
    "name": "X-Hub-Signature"
  }
}
```

To avoid the chicken and egg problem, I would not use `hooks.EnvNamespace` within `getenv`.

**Comment by adnanh** [comment]
That would be awesome.

**Comment by jamielennox** [comment]
That would absolutely work for me.

**Comment by moorereason** [comment]
Fixed by #161

**Comment by moorereason** [comment]
This can be closed now.
